docs/credentials.adoc

```diff
@@ -1,16 +1,135 @@
-= cerdentials.zip file format
+= credentials.zip file format
 
-To authenticate on the ostree server, aktualizr uses cerdentials.zip archive.
+The credentials.zip file provides the link between a user's on-line account on ATS Garage or OTA+ deployment and developer's machine. It is used for:
 
-The minimal requirement for this archive is including `treehub.json` file with
-the following content:
+* Uploading the built OSTree image to treehub with garage-push
+* Offline-signing Uptane images targets metadata with garage-sign
+* Provisioning the device with Aktualizr or aktualizr_cert_provider
+* Providing the URL to the update server to Aktualizr (for implicit provisioning this is templated into sota.toml at build time, to allow software updates to modify that URL)
 
+The following files are present in credentials.zip:
+
+[options="header"]
+|======================
+| Filename in zip | Purpose | Used by
+| treehub.json | Location and authentications for treehub and Uptane repo | garage-sign, garage-push
+| client.crt | Certificate for TLS client authentication | garage-push
+| client.key | Private key for TLS client authentication | garage-push
+| autoprov_credentials.p12 | TLS client credentials for automatic device provisioning | aktualizr, aktualizr_cert_provider
+| autoprov.url | URL for automatic provisioning server | aktualizr, aktualizr_cert_provider
+| root.json | Initial Uptane root.json (for secure bootstrapping) | garage-sign
+| targets.pub | Public key for offline Uptane image signing | garage-sign
+| targets.sec | Private key for offline Uptane image signing | garage-sign
+| tufrepo.url | URL to Uptane repository | garage-sign
+| root.crt | Root CA for TLS client authentication | aktualizr
```

Comment:
Is there ever a time that we have `client.crt`, `client.key`, or `root.crt` separate from `autoprov_credentials.p12`?

Reply:
Currently implicit_writer uses the `root.crt` extracted from `autoprov_credentials.p12` as well as `autoprov.url`.

Reply:
I think that `client.crt` `client.key` and `root.crt` are only used by garage-push and only for `certificate_auth`

Reply:
Yeah, but my point is just that the are actually just contained with the p12 file, they aren't included separately.

Reply:
(Patrick and I talked about this face to face)